tawazi/dag.py

```diff
@@ -995,9 +1005,9 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVDAG:
             RVDAG: the return value of the DAG's Execution
         """
         if self.executed:
-            raise TawaziUsageError(
-                "DAGExecution object should not be reused. Instantiate a new one"
-            )
+            warnings.warn("DAGExecution object's reuse is not recommended.", stacklevel=2)
```

Comment:
because it is easier for a user who is just testing or doing a forloop to reuse the same DAGExecution rather than create a new one... this saves him a line of code.
This should be farely easy to support too